src/expr/udf/src/external.rs

```diff
@@ -101,6 +109,36 @@ impl ArrowFlightUdfClient {
 
     /// Call a function.
     pub async fn call(&self, id: &str, input: RecordBatch) -> Result<RecordBatch> {
+        let metrics = &*GLOBAL_METRICS;
+        let labels = &[self.addr.as_str(), id];
+        metrics
+            .udf_input_chunk_rows
+            .with_label_values(labels)
+            .observe(input.num_rows() as f64);
```

Comment:
Appears that we may leverage the `label_guarded` metrics from @wenym1 here. 👀

Reply:
What are its advantages over the original prometheus API? 👀

Reply:
It will automatically remove the label from the metric vec when the label is not used anymore. But we cannot simply replace the raw metric vec with label guarded one in this PR because in this PR we drop the value returned from `with_label_values` immediately after using it, which may reset and reinitialize the metric.

I will clear all similar usages when resolving https://github.com/risingwavelabs/risingwave/issues/13981 and https://github.com/risingwavelabs/risingwave/issues/12805.